src/tasks/minions/minigames/giantsFoundryActivity.ts

```diff
@@ -0,0 +1,109 @@
+import { deepClone } from 'e';
+import { Bank } from 'oldschooljs';
+
+import {
+	encodeGiantWeapons,
+	generateRandomGiantWeapon,
+	GiantsFoundryBank,
+	giantWeaponName
+} from '../../../lib/giantsFoundry';
+import { trackLoot } from '../../../lib/lootTrack';
+import { incrementMinigameScore } from '../../../lib/settings/settings';
+import { SkillsEnum } from '../../../lib/skilling/types';
+import { GiantsFoundryActivityTaskOptions } from '../../../lib/types/minions';
+import { randomVariation } from '../../../lib/util';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import { updateBankSetting } from '../../../lib/util/updateBankSetting';
+import { giantsFoundryAlloys } from '../../../mahoji/lib/abstracted_commands/giantsFoundryCommand';
+
+export const giantsFoundryTask: MinionTask = {
+	type: 'GiantsFoundry',
+	async run(data: GiantsFoundryActivityTaskOptions) {
+		const { quantity, userID, channelID, duration, metalScore, alloyID } = data;
+		const user = await mUserFetch(userID);
+		const alloy = giantsFoundryAlloys.find(i => i.id === alloyID);
+		if (!alloy) {
+			return 'An issue occurred trying to find a alloy using the alloy ID.';
+		}
```

Comment:
you can null assert in cases like this. remove this `if`, and use `!` to say its always there. we want it to error if its not there